src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/TextDecorationCollectionConverter.cs

```diff
@@ -2,285 +2,159 @@
-                    );
+            // Flags indicating which pre-defined TextDecoration have been matched
+            byte matchedDecorations = 0;
 
-                    if (i < TextDecorationNames.Length)
-                    {
-                        // Found a match within the predefined names
-                        if ((MatchedTextDecorationFlags & (1 << i)) > 0)
-                        {
-                            // Error: The matched value is duplicated.
-                            index = -1;
-                        }
-                        else
-                        {
-                            // Valid match. Add to the collection and remember that this text decoration
-                            // has been added
-                            textDecorations.Add(PredefinedTextDecorations[i]);
-                            MatchedTextDecorationFlags |= (byte)(1 << i);
+            // Sanitize the input
+            ReadOnlySpan<char> decorationsSpan = text.AsSpan().Trim();
 
-                            // Advance to the start of next name
-                            index = AdvanceToNextNameStart(text, index + TextDecorationNames[i].Length);                            
-                        }
-                    }
-                    else
-                    {
-                        // Error: no match found in the predefined names
-                        index = -1;
-                    }
-                }
-            }
+            // Test for "None", which equals to empty collection and needs to be specified alone
+            if (decorationsSpan.IsEmpty || decorationsSpan.Equals("None", StringComparison.OrdinalIgnoreCase))
+                return new TextDecorationCollection();
 
-            if (index < 0)
-            {
-                throw new ArgumentException(SR.Format(SR.InvalidTextDecorationCollectionString, text));
-            }
-            
-            return textDecorations;            
-        }        
+            // Create new collection, save re-allocations
+            TextDecorationCollection textDecorations = new(1 + decorationsSpan.Count(','));
 
-        /// <summary>
-        /// ConvertTo
-        /// </summary>
-        /// <param name="context"> ITypeDescriptorContext </param>
-        /// <param name="culture"> CultureInfo </param>        
-        /// <param name="value"> the object to be converted to another type </param>
-        /// <param name="destinationType"> The destination type of the conversion </param>
-        /// <returns> null will always be returned because TextDecorations cannot be converted to any other type. </returns>        
-        public override object ConvertTo(ITypeDescriptorContext context, CultureInfo culture, object value, Type destinationType)
-        {
-            if (destinationType == typeof(InstanceDescriptor) && value is IEnumerable<TextDecoration>)
+            // Go through each item in the input and match accordingly
```

Comment:
This is my personal opinion, feel free to take it or leave it:

I've noticed in your PRs that you tend to add a lot of comments to explain what your code does, essentially duplicating the logic in the code and in the comment. These types of comment tend to make it harder to read the code, modify the code and they often become outdated when the code changes. Comments are most useful when explaining why the code does what it does, adding a reference to something (Like an issue or web link) or explaining something that is not easily understandable by reading the code.

If you want to read more about this, I would suggest this blog post which goes into more details: https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/

Your PR doesn't seem to go against any of the coding guidelines so at the end of the day it's up to the WPF team to decide if you need to change anything in your PR.

Again, this is my personal opinion and I won't pretend I'm the absolute authority when it comes to code comments. This is merely a suggestion since I don't know your background and I don't know if you might find this suggestion useful.

Reply:
Thanks Thomas, I agree this one for example isn't that relevant and just duplicates the whole proc.

Working with large codebases especially written in C and some still in assembly, same as largely working with disassembly (not everyone gets the beauty of working with VS, the debugger its tooling), you start to comment way more even the more obvious things to follow the train of thought.

Often times, when fixing bugs, debugging, few years after someone, you'll find that the most important thing is not the code written, but the intention behind one programmer's code (which often is just a comment, since they left). People make mistakes, today's CRs and collaboration over one's code are way more prevalent than they used to be (at least in my experience), and when you don't have that clue, it is harder to debug and fix such code should there be a bug. Does maintability in case of any refactoring hurt? Yes, you have to take care of the comment. Does it help you to make more informed decision? It does.

Then again, we're in .NET when most things are way more expressive and it is easier to go through code itself and understand the intent because multiple things are abstracted away, so less comments like this are needed. When I do these refactorings on whole class, I just write down comments on stuff I shouldn't forget and then just run with it, and yes, sometimes I do not delete them ("add" them instead).

So it is indeed something I could improve on when swapping to .NET each time. Habits die hard.